ast.c

```diff
@@ -1037,6 +1071,11 @@ static void ast_to_zval(zval *zv, zend_ast *ast, ast_state_info_t *state) {
 		// This is an undeclared dynamic property and has no cache slot.
 		ast_update_property_long(zv, AST_STR(str_endLineno), decl->end_lineno);
 
+#if PHP_VERSION_ID >= 80400
+		if (kind == ZEND_AST_CLOSURE || kind == ZEND_AST_ARROW_FUNC) {
+			ZVAL_STR(&tmp_zv, AST_STR(str_bracketed_closure));
```

Comment:
What's this change for?

Reply:
It's because of https://github.com/php/php-src/pull/13550/files#diff-3e6742a9069b5717cf961c9d6b2aefbd1c730869d8a58123b1b5f3bc3e9082fc - php 8.4-dev has the name of closures be null instead of the old literal string `{closure}`.

This affects unit tests, but shouldn't really affect phan.

This could probably be extended to be consistently null in versions 110+ of php-ast.

Reply:
Or have the field be omitted entirely for closures and arrow functions.

Reply:
Probably omitting it entirely would be best. But we can leave that for a followup.